R/CITEseq.R

```diff
@@ -0,0 +1,108 @@
+#'     
+#'     Available datasets are:
+#'     \itemize{
+#'         \item{cord_blood: } a dataset of single cells of cord blood as 
+#'         provided in Stoeckius et al. (2017).
+#'          \itemize{
+#'             \item{scRNA_Counts} - Stoeckius scRNA-seq gene count matrix
+#'             \item{scADT} - Stoeckius antibody-derived tags (ADT) data
+#'             }
+#'      }
+#'      
+#' @param DataType character(1) indicating the identifier of the dataset to
+#'     retrieve.  (default "cord_blood")
+#'
+#' @param modes character( ) The assay types or modes of data to obtain these
+#'     include scADT and scRNA-seq data by default.
+#' @param version character(1) Either version '1.0.0' depending on
+#'     data version required.
+#' @param dry.run logical(1) Whether to return the dataset names before actual
+#'     download (default TRUE)
+#'
+#' @param verbose logical(1) Whether to show the dataset currently being
+#'     (down)loaded (default TRUE)
+#'
+#' @param ... Additional arguments passed on to the
+#'     \link[ExperimentHub]{ExperimentHub-class} constructor
+#'
+#' @return A single cell multi-modal \linkS4class{MultiAssayExperiment} or
+#'     informative `data.frame` when `dry.run` is `TRUE`
+#' @references Stoeckius et al. (2017)
+#' @export
+#'
+#' @examples
+#' mse <- CITEseq(dry_run=FALSE)
+#' experiments(mse)
+CITEseq <- function(DataType="cord_blood", modes="*", 
+                    version="1.0.0", dry.run=TRUE, verbose=TRUE, ...)
+{
+    
+    ess_list <- .getResourcesList(prefix = "citeseq_", datatype = DataType,
+                                modes=modes, version=version, 
+                                dry.run=dry.run,
+                                verbose=verbose)#, ...)
+    if(!dry.run)
+    {
+    
+        switch (DataType,
+                "cord_blood"={
+                    mse <- .cord_blood(ess_list=ess_list)
+                },
+                ## Add here other CITE-seq datasets based on DataType identifier
+                {
+                    stop("Unrecognized CITE-seq dataset name!")
+                }
+        )
+        return(mse)
+    } else {
+        return(ess_list)
+    }
+    
+    
+}
+
+
+#' CITEseqMseToSce
+#' @description converts a MultiAssayExperiment object with CITEseq data into
+#' a SingleCellExperiment object to be used with already known methods and 
+#' packages in literature.
+#' 
+#' @param mse a MultiAssayExperiment object with scRNA and scADT named experiments
+#'
+#' @return a SingleCellExperiment object as widely with scRNA data as counts and 
+#' scADT data as altExps
+#' @importFrom MultiAssayExperiment experiments
+#' @importFrom SummarizedExperiment SummarizedExperiment
+#' @importFrom SingleCellExperiment SingleCellExperiment
+#' @export
+#'
+#' @examples
+#' mse <- CITEseq(dry_run=FALSE)
+#' sce <- CITEseqMseToSce(mse)
+#' 
+CITEseqMseToSce <- function(mse)
+{
+    stopifnot(is(mse, "MultiAssayExperiment"))
```

Comment:
You also need to add the 'methods' package to the 'DESCRIPTION' when you're using the  'is' function.

Reply:
and the import directive from 'roxygen' '@importFrom methods is'